tools/autoenv.sh

```diff
@@ -1,38 +1,46 @@
-#!/bin/bash
+#!/bin/sh
 
 # This script attempts to detect the correct environmental variables required
 # to build JSS from source.
 
-# Load OS information.
-if [ -f /etc/os-release ]; then
-    . /etc/os-release
-elif [ -f /usr/lib/os-release ];then
-    . /usr/lib/os-release
-else
-    echo "Cannot determine OS information. Exiting..." 1>&2
-    exit 1
-fi
-
 # Try and detect a JDK installation.
-if [ -d "/etc/alternatives/java_sdk" ]; then
+if [ "x$JAVA_HOME" = "x" ] && [ -e "/etc/alternatives/java_sdk" ]; then
     export JAVA_HOME="/etc/alternatives/java_sdk"
-else
+fi
+if [ "x$JAVA_HOME" = "x" ]; then
```

Comment:
I decided to split these into separate `if ... ; then ... else` blocks versus chaining them.